solution.rb

```diff
@@ -201,23 +204,63 @@
 ]
 
 most_expensive_products_by_store = []
-qw = []
+#loop in stores
 stores.each do |store|
-    price_a = []
-    hi_price = 0.0
-    
+    #for each store create these==
+    price_array = []
+    description_array = []
+    name_of_higer_product = ""
+    hiegher_price = 0.0
+    #loop inside each product inside stores
     store[:products].map do |product|
+        #gitting  prices and  names
         all_price = product[:price]
-        price_a.push(all_price.to_f)
+        prudct_name = product[:description]
+        #pushing price to price array
+        price_array.push(all_price.to_f)
+        #pushing product name to description array
+        description_array.push(prudct_name)
+    end # end of inner loop
+    #check what price is higher and make it the heigher price and get the product name as well
+    if price_array[0] > price_array[1]
+        hiegher_price = price_array[0]
+        name_of_higer_product = description_array[0]
+    else
+        hiegher_price = price_array[1]
+        name_of_higer_product = description_array[1]
     end
+    #push the result to the givin array
+    most_expensive_products_by_store.push({store_name: store[:store_name], most_expensive_product: {description: name_of_higer_product, price: hiegher_price}})
     
-    if price_a[0] > price_a[1]
-        hi_price = price_a[0]
+end
 
-    else
-        hi_price = price_a[1]
-    end
+#===========================
+#===========================
 
-    qw.push({store_name: store[:store_name], most_expensive_product: {price: hi_price}} )
```

Comment:
Consider this solution, it is based on yours. It's minimizing the number of arrays you use, this still needs to be modified to accommodate more than 2 products per store
```ruby
most_expensive_products_by_store = stores.map do |store|
  
   products = store[:products].map do |product|
        { price: product[:price], name: product[:description]}
    end 

    if products[0][:price] > products[1][:price]
        {store_name: store[:store_name], price: products[0][:price], most_expensive_product: products[0][:name]}
    else
        { store_name: store[:store_name],price: products[1][:price], most_expensive_product: products[1][:name]}        
    end
  
end
```